packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -210,4 +217,70 @@ export class MetadataOrchestrator {
         return 'string';
     }
   };
+
+  public eligibilityDelegate = async (
+    requests: ApexOASEligiblePayload,
+    maxRetries: number = 3
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    let response = {};
+    let attempt = 0;
+    const languageClient = languageClientUtils.getClientInstance();
+    if (languageClient) {
+      while (attempt < maxRetries) {
+        try {
+          attempt++;
+          response = await languageClient?.sendRequest('apexoas/isEligible', requests);
+          telemetryService.sendEventData('isEligibleResponseSucceeded', {
```

Comment:
What other telemetry info is worth capturing here?

Reply:
Regarding the request payload, I think maybe we can capture if the command is for folders or just one class/classes. But I am not sure if we want to capture the info here compared to the command dispatch function, since the request does not have to be successful. But I will add that here first.

I don't think we need any info worth capturing in the response, since the info is about the eligibility of each method. not about the use case.